File: CompositePattern/CompositePattern/Components/Menu.cs
using CompositePattern.Iterators;

namespace CompositePattern.Components;

public class Menu : MenuComponent
{
    public List<MenuComponent> Components { get; set; }
    
    public string Name { get; set; }
    
    public string Description { get; set; }

    public Menu(string name, string description)
    {
        Components = new List<MenuComponent>();
        Name = name;
        Description = description;
    }
    
    public override void Add(MenuComponent component)
    {
        Components.Add(component);
    }

    public override void Remove(MenuComponent component)
    {
        Components.Remove(component);
    }

    public override MenuComponent GetChild(int i)
    {
        return Components[i];
    }

    public override void Print()
    {
        Console.WriteLine("Name: " + Name);
        Console.WriteLine("Description: " + Description);
        Console.WriteLine("============");

        IIterator iterator = CreateIterator();
        while (iterator.HasNext())
        {
            var component = iterator.Next() as MenuComponent;
            component?.Print();
        }
        
    }
    
    public override IIterator CreateIterator()
    {
        return new MenuComponentIterator(Components);
    } 
}

File: CompositePattern_Java/src/main/java/Components/Menu.java
package Components;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class Menu extends MenuComponent
{
    public List<MenuComponent> Components;

    public String Name;

    public String Description;

    public Menu(String name, String description)
    {
        Components = new ArrayList<>();
        Name = name;
        Description = description;
    }

    @Override
    public void Add(MenuComponent component)
    {
        Components.add(component);
    }

    @Override
    public void Remove(MenuComponent component)
    {
        Components.remove(component);
    }

    @Override
    public MenuComponent GetChild(int i)
    {
        return Components.get(i);
    }

    @Override
    public void Print()
    {
        System.out.println("Name: " + Name);
        System.out.println("Description: " + Description);
        System.out.println("============");

        Iterator iterator = CreateIterator();
        while (iterator.hasNext())
        {
            var component = (MenuComponent) iterator.next();
            component.Print();
        }
    }

    @Override
    public Iterator CreateIterator()
    {
        return Components.iterator();
    }
}
